README.md

```diff
@@ -127,3 +127,205 @@ Then translated to
++}
+```
+
+Then, conditionally create a `redaction` node instead of the desired regular
+node when in redaction mode (see more about the `redaction` node
+[here](https://github.com/code-dot-org/redactable-markdown/blob/29c64b3c9e736e28746e6a8627b57c8cc3f0dcc0/src/plugins/process/restorationRegistration.js#L8-L15)):
+
+Note here that all that is required of the redaction node is that it contains a
+unique `redactionType` identifier, and any information required to recreate the
+node.
+
+```diff
+diff --git a/mention.js b/mention.js
+index 7a6fc91..08f1bf0 100644
+--- a/mention.js
++++ b/mention.js
+@@ -1,10 +1,15 @@
+ module.exports = mentions;
+
++var redact;
++
+ function mentions() {
+   var Parser = this.Parser;
+   var tokenizers = Parser.prototype.inlineTokenizers;
+   var methods = Parser.prototype.inlineMethods;
+
++  /* Make the Parser's redact option visible to the tokenizer */
++  redact = Parser.prototype.options.redact;
++
+   /* Add an inline tokenizer (defined in the following example). */
+   tokenizers.mention = tokenizeMention;
+
+@@ -24,7 +29,19 @@ function tokenizeMention(eat, value, silent) {
+     }
+
+     var add = eat(match[0]);
+-    return createMention(add, match[1], match[0]);
++    var name = match[1];
++    var text = match[0];
++
++    if (redact) {
++      return add({
++        type: 'redaction',
++        redactionType: 'mention',
++        name: name,
++        text: text
++      });
++    }
++
++    return createMention(add, name, text);
+   }
+ }
+```
+
+Finally, add a restoration method for the specified redaction type, using the
+newly-isolated node creation logic.
+
+```diff
+diff --git a/mention.js b/mention.js
+index 08f1bf0..beb01ca 100644
+--- a/mention.js
++++ b/mention.js
+@@ -6,6 +6,11 @@ function mentions() {
+   var Parser = this.Parser;
+   var tokenizers = Parser.prototype.inlineTokenizers;
+   var methods = Parser.prototype.inlineMethods;
++  var restorationMethods = Parser.prototype.restorationMethods;
++
++  restorationMethods.mention = function (add, node) {
++    return createMention(add, node.name, node.text);
++  }
+
+   /* Make the Parser's redact option visible to the tokenizer */
+   redact = Parser.prototype.options.redact;
+```
+
+We can now redact and restore `@` mentions:
+
+```bash
+$ echo "Hello @example" > source.md
+$ redact source.md -p mention.js | tee redacted.md
+Hello [][0]
+$ sed 's/Hello/Bonjour/' redacted.md | tee translated.md
+Bonjour [][0]
+$ restore -s source.md -r translated.md -p mention.js
+Bonjour [@example](https://social-network/example)
+```
+
+### Adanced Redaction Example
+
+We also have the option of allowing the redaction and restoration process to
+change the way the parsed text is processed.
+
+Say we wanted the redacted version of the basic example to look like:
+
+    Hello [example][0]
+
+And for changes made to the text in the redaction to be reflected in the URL
+like:
+
+    Bonjour [exemple][0] > Bonjour [@example](https://social-network/exemple)
```

Comment:
It's a little unintuitive, here and in the "results" bash block below, that the visible text of the mention does not change from `example` to `exemple` but the URL does.  I'm not sure what change I'm suggesting in its place, but it took me a while to grok this section because I was wondering why "example" didn't get translated.

Reply:
Oh, good point. I'll switch which piece of content is the one that gets translated